### PR TITLE
Update ipartition to 3.5.1

### PR DESCRIPTION
**Pull request by colindunn**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` left no offenses.
- [x] The commit message includes the cask’s name and version.

### CODE REVIEW

File: Casks/ipartition.rb [hunk 1]
```diff
@@ -1,8 +1,16 @@
 cask 'ipartition' do
-  version '3.4.5'
-  sha256 '0b66c44dfb4b8525056b68bfd8f4d56fe4b78fd0e5e4705659bfd242e2722319'
+  if MacOS.version < '10.10'
```
Comment (miccal) [inline Casks/ipartition.rb]: Please change this to `if MacOS.version <= :mavericks` for consistency with other Casks.

Comment (colindunn) [inline Casks/ipartition.rb]: Done

Comment (miccal) [inline Casks/ipartition.rb]: It needs to be `<=`, not just `<`.

Comment (miccal) [inline Casks/ipartition.rb]: It needs to be `<=`, not just `<`.

Comment (miccal) [inline Casks/ipartition.rb]: Thank you.

**Review by miccal — APPROVED** [review]

**Pull request merged**